src/boot/efi/stub.c

```diff
@@ -222,7 +237,15 @@ EFI_STATUS efi_main(EFI_HANDLE image, EFI_SYSTEM_TABLE *sys_table) {
 
         export_variables(loaded_image);
 
-        (void) pack_cpio(loaded_image,
+        loaded_image_root = LibOpenRoot(loaded_image->DeviceHandle);
+        if (!loaded_image_root)
+                return log_error_status_stall(EFI_LOAD_ERROR, L"Unable to open root directory.");
```

Comment:
When this fails, it causes the whole boot to fail instead of just the initrd drop-in. Is that alright?

Reply:
True, but it's rather unlikely that this one would fail. But I guess you could just keep the path assembling inside of `pack_cpio` and just pass the global path as an arg (which may be NULL to indicate the path should be constructed as before). Should've suggested that from the start. :D

Reply:
:shrug: I've come to think that computing the path outside of `pack_cpio makes more sense conceptually. I could put the path generation + pack_cpio calls into their own function next to combine_initrd. Thoughts on that?

Reply:
I could do it the way you propose also if you'd prefer that, of course!

Reply:
I'd prefer the smaller patch. :D

Reply:
Done